packages/block-editor/src/components/inserter/pattern-panel.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * WordPress dependencies
+ */
+import { Dropdown, MenuGroup, MenuItem, Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { check } from '@wordpress/icons';
+
+function PatternInserterPanel( {
+	title,
+	selectedCategory,
+	patternCategories,
+	onClickCategory,
+	children,
+} ) {
+	const renderToggle = ( { isOpen, onToggle } ) => (
+		<h2 className="block-editor-inserter__panel-title">
+			{ __( 'Viewing' ) }:
+			<Button onClick={ onToggle } aria-expanded={ isOpen }>
+				{ title ? title : selectedCategory.label }
+			</Button>
+		</h2>
+	);
+
+	const renderContent = ( { onClose } ) => {
+		return (
+			<MenuGroup label={ __( 'Select a Category' ) }>
+				{ patternCategories.map( ( patternCategory ) => {
+					return (
+						<MenuItem
+							key={ patternCategory.name }
+							onClick={ () => {
+								onClickCategory( patternCategory );
+								onClose();
+							} }
+							className={
+								patternCategory.name === selectedCategory.name
+									? 'is-selected'
+									: ''
+							}
+							icon={ check }
+						>
+							{ patternCategory.label }
+						</MenuItem>
+					);
+				} ) }
+			</MenuGroup>
+		);
+	};
+
+	return (
+		<>
+			<div className="block-editor-inserter__panel-header">
+				<Dropdown
```

Comment:
I wonder if this should use `CustomSelectControl` instead as this component has already been reviewed properly in terms of a11y... cc @enriquesanchez

Reply:
Is it possible to get the same level of styling on each item and to the control itself? I can give it a go and see.

Reply:
Why would we want to give it custom styling?

Reply:
I see `CustomSelectControl` has already been implemented, so 👍 👍  from me